subworkflows/local/prepare_input_stitch/prepare_input_stitch.nf

```diff
@@ -0,0 +1,64 @@
+include { BCFTOOLS_QUERY                    } from '../../../modules/nf-core/bcftools/query/main'
+include { BCFTOOLS_NORM                     } from '../../../modules/nf-core/bcftools/norm/main'
+include { GAWK                              } from '../../../modules/nf-core/gawk'
+
+
+workflow PREPARE_INPUT_STITCH {
+
+    take:
+    ch_panel_sites
+    ch_fasta
+    ch_input_impute
+
+    main:
+
+    ch_versions      = Channel.empty()
+
+    // Prepare posfile and parameters for STITCH
+    // Convert position file to tab-separated file
+    BCFTOOLS_QUERY(ch_panel_sites, [], [], [])
+    ch_posfile = BCFTOOLS_QUERY.out.output
+
+    // Remove multiallelic positions from tsv
+    GAWK(ch_posfile, [])
+
+    // Get chromosomes of posfile
+    ch_posfile = GAWK.out.output.map{meta, posfile -> return[['chr': meta.chr], posfile]}
+
+    // Get chromosomes of fasta
+    ch_chromosomes = ch_fasta.map{it -> it[2]}
+                    .splitCsv(header: ["chr", "size", "offset", "lidebase", "linewidth", "qualoffset"], sep: "\t")
+                    .map{it -> return [[chr: it.chr], it.chr]}
+
+    // Combine channels
+    def input_empty         = [[]]
+    def rdata_empty         = [[]]
+    k_val                   = params.k_val
+    ngen                    = params.ngen
+
+    // Make final channel with parameters
+    stitch_parameters = ch_posfile.map { it + input_empty + rdata_empty}
```

Comment:
If I understand well this is also pre-processing of the panel ?

Reply:
This is not preprocessing, but part of the input channel preparation for stitch.